vars/cloudFoundryDeploy.groovy

```diff
@@ -319,14 +336,27 @@ def deployMta (config) {
         usernameVariable: 'username'
     )]) {
         echo "[${STEP_NAME}] Deploying MTA (${config.mtaPath}) with following parameters: ${config.mtaExtensionDescriptor} ${config.mtaDeployParameters}"
+        def cfTraceFile = 'cf.log'
         def returnCode = sh returnStatus: true, script: """#!/bin/bash
             export HOME=${config.dockerWorkspace}
+            export CF_TRACE="${cfTraceFile}"
             set +x
             set -e
             cf api ${config.cloudFoundry.apiEndpoint}
             cf login -u ${username} -p '${password}' -a ${config.cloudFoundry.apiEndpoint} -o \"${config.cloudFoundry.org}\" -s \"${config.cloudFoundry.space}\"
             cf plugins
             cf ${deployCommand} ${config.mtaPath} ${config.mtaDeployParameters} ${config.mtaExtensionDescriptor}"""
+        if(config.verbose || returnCode != 0) {
+            if(fileExists(file: cfTraceFile)) {
```

Comment:
Quite much code duplication. I think we could use a helper method. Might also be possible withEnv instead so that we do not have to modify the sh call above.

Reply:
I already commented above:

>Next step (in another PR): unify duplicate coding. We have two ways how the deployment can happen. Currently we have different coding for each way.